src/main/java/Storage.java

```diff
@@ -0,0 +1,157 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles loading and saving of tasks to/from file storage.
+ */
+public class Storage {
+    private final String filePath;
+    private final String directoryPath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        // Extract directory path from file path
+        Path path = Paths.get(filePath);
+        this.directoryPath = path.getParent() != null ? path.getParent().toString() : "";
+    }
+
+    /**
+     * Loads tasks from file. Returns empty list if file doesn't exist or has errors.
+     */
+    public List<Task> loadTasks() throws KiwiException {
+        List<Task> tasks = new ArrayList<>();
+
+        try {
+            createDirectoryIfNotExists();
+
+            if (!Files.exists(Paths.get(filePath))) {
+                return tasks;
+            }
+
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                try {
+                    Task task = parseTaskFromString(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                } catch (Exception e) {
+                    System.out.println("Warning: Skipping corrupted line: " + line);
+                }
+            }
+            reader.close();
+
+        } catch (IOException e) {
+            throw new KiwiException("Error loading tasks from file: " + e.getMessage());
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Saves tasks to file.
+     */
+    public void saveTasks(List<Task> tasks) throws KiwiException {
+        try {
+            createDirectoryIfNotExists();
+
+            BufferedWriter writer = new BufferedWriter(new FileWriter(filePath));
+
+            for (Task task : tasks) {
+                writer.write(convertTaskToString(task));
+                writer.newLine();
+            }
+
+            writer.close();
+
+        } catch (IOException e) {
+            throw new KiwiException("Error saving tasks to file: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Creates directory if it doesn't exist.
+     */
+    private void createDirectoryIfNotExists() throws IOException {
+        if (!directoryPath.isEmpty()) {
+            Files.createDirectories(Paths.get(directoryPath));
+        }
+    }
+
+    /**
+     * Converts a task to string format for storage.
+     * Format: TaskType | isDone | description | additionalInfo
+     */
+    private String convertTaskToString(Task task) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
```suggestion
        StringBuilder strBuilder = new StringBuilder();
```
Maybe use a slightly less ambiguous name?